File: Offer/Java/Offer03.java
import java.util.HashSet;
import java.util.Iterator;
import java.util.Set;

public class Offer03 {

    public int findRepeatNumber(int[] nums) {

        Set<Integer> set = new HashSet<>();
        for(int num:nums){
            if(set.contains(num)){
                return num;
            }
            set.add(num);
        }
        return -1;
    }
}

File: Offer/C++/offer03.cc
#include<iosteam>
#include<vector>
#include<unordered_map>
#include<algorithm>
using namespace std;

class Solution {
public:
    int findRepeatNumber(vector<int>& nums) {
        // if(nums.size() == 0){
        //     return 0;
        // }
        // sort(nums.begin(),nums.end());

        // int l = 0;
        // int r = 1;
        
        // for(int i = 0; i < nums.size(); i++){
        //     if(nums[i] == nums[r]){
        //         return nums[i];
        //     }
        //     l++;
        //     r++;
        // }
        // return 0;

    unordered_map<int,bool> map;
    for(int num : nums){
            if(map[num]){
                return num;
            }
            map[num] = true;
        }

        return -1;
    }
};